tests/test_apispec_oneofschema.py

```diff
@@ -0,0 +1,140 @@
+import pytest
+from apispec import APISpec
+from marshmallow import Schema, fields
+from marshmallow_oneofschema import OneOfSchema
+from apispec_oneofschema import MarshmallowPlugin
+
+class TreeSchema(Schema):
+    leaves = fields.Int(required=True)
+
+class FlowerSchema(Schema):
+    blooming = fields.Bool(required=True)
+
+class PlantSchema(OneOfSchema):
+    type_schemas = {
+        'tree': TreeSchema,
+        'flower': FlowerSchema
+    }
+
+class ForestSchema(Schema):
+    plant = fields.Nested(PlantSchema)
+
+@pytest.fixture()
+def spec():
+    return APISpec(
+        title='Botany',
+        version='1.0.0',
+        openapi_version='3.0.0',
+        plugins=[
+            MarshmallowPlugin(),
+        ]
+    )
+
+def test_top_level(spec):
+    spec.components.schema('Plant', schema=PlantSchema)
+
+    schemas = spec.to_dict()['components']['schemas']
+    plant = schemas['Plant']
+    discriminator = plant['discriminator']
+    oneof = plant['oneOf']
+
+    assert discriminator['mapping']['flower'] ==  '#/components/schemas/Flower'
+    assert discriminator['propertyName'] == 'type'
+    assert {'$ref': '#/components/schemas/Flower'} in oneof
+    assert {'$ref': '#/components/schemas/Tree'} in oneof
+
+    assert 'Tree' in schemas
+    assert 'Flower' in schemas
+
+
+def test_nested(spec):
+    spec.components.schema('Forest', schema=ForestSchema)
+
+    schemas = spec.to_dict()['components']['schemas']
+
+    forest = schemas['Forest']
+    assert forest['properties']['plant'] == {'$ref': '#/components/schemas/Plant'}
+
+    plant = schemas['Plant']
+    discriminator = plant['discriminator']
+    oneof = plant['oneOf']
+
+    assert discriminator['mapping']['flower'] ==  '#/components/schemas/Flower'
+    assert discriminator['propertyName'] == 'type'
+    assert {'$ref': '#/components/schemas/Flower'} in oneof
+    assert {'$ref': '#/components/schemas/Tree'} in oneof
+
+    assert 'Tree' in schemas
+    assert 'Flower' in schemas
+
+
+def test_rsolver_returns_none():
```

Comment:
s/test_rsolver_returns_none/test_resolver_returns_none

Reply:
Fixed in 9d9ab7036514a0e8f8fce15fa0f0cd938560dc52.